unraid_mcp/tools/storage.py

```
@@ -101,17 +90,22 @@ async def unraid_storage(
           logs - Retrieve log content (requires log_path, optional tail_lines)
         """
-        if action not in QUERIES:
-            raise ToolError(f"Invalid action '{action}'. Must be one of: {list(QUERIES.keys())}")
+        if action not in ALL_ACTIONS:
+            raise ToolError(f"Invalid action '{action}'. Must be one of: {sorted(ALL_ACTIONS)}")
 
         if action == "disk_details" and not disk_id:
             raise ToolError("disk_id is required for 'disk_details' action")
 
+        if tail_lines < 1 or tail_lines > _MAX_TAIL_LINES:
+            raise ToolError(f"tail_lines must be between 1 and {_MAX_TAIL_LINES}, got {tail_lines}")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)